wolfcrypt/src/port/atmel/atmel.c

```diff
@@ -915,62 +929,72 @@ int atcatls_verify_signature_cb(WOLFSSL* ssl, const byte* sig, unsigned int sigS
+    size_t signerCertSize=0;
+    size_t deviceCertSize=0;
+    uint8_t *certBuffer;
 
-    /*Read signer cert*/
-    status = tng_atcacert_read_signer_cert(&certBuffer[ATCATLS_TNGTLS_DEVICE_CERT_SIZE],
-                                           &signerCertSize);
+    /* fetch signer cert size */
+    status=tng_atcacert_read_signer_cert(NULL, &signerCertSize);
     if (ATCA_SUCCESS != status) {
-        ret = atmel_ecc_translate_err(ret);
-        return ret;
-    }
-    if (signerCertSize != ATCATLS_TNGTLS_SIGNER_CERT_SIZE) {
         #ifdef WOLFSSL_ATECC_DEBUG
-        printf("signer cert size != ATCATLS_TNGTLS_SIGNER_CERT_SIZE.(%d)\r\n",
-               signerCertSize);
+        printf("Failed reading Signer cert size(0x%x)\r\n", status);
         #endif
         return WOLFSSL_FAILURE;
     }
 
-    /*Read device cert signed by the signer above*/
-    status = tng_atcacert_read_device_cert(certBuffer, &deviceCertSize,\
-     	                  &certBuffer[ATCATLS_TNGTLS_DEVICE_CERT_SIZE]);
+    /* fetch device cert size */
+    status=tng_atcacert_read_device_cert(NULL, &deviceCertSize, NULL);
     if (ATCA_SUCCESS != status) {
+        #ifdef WOLFSSL_ATECC_DEBUG
+        printf("Failed reading device cert size(0x%x)\r\n", status);
+        #endif
+        return WOLFSSL_FAILURE;
+    }    
+    certBuffer=(uint8_t*)XMALLOC(signerCertSize+deviceCertSize, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    if(NULL == certBuffer){
+        #ifdef WOLFSSL_ATECC_DEBUG
+        printf("Failed allocating space for certBuffer\r\n");
+        #endif
+        return WOLFSSL_FAILURE;
+    }
+    
+    /* Read signer cert */
+    status = tng_atcacert_read_signer_cert(&certBuffer[deviceCertSize],
+                                           &signerCertSize);
+    if (ATCA_SUCCESS != status) {
+        #ifdef WOLFSSL_ATECC_DEBUG
+        printf("Error reading signer cert(0x%x)\r\n", status);
+        #endif
+        XFREE(certBuffer,NULL,DYNAMIC_TYPE_TMP_BUFFER);
         ret = atmel_ecc_translate_err(ret);
         return ret;
     }
-    if (deviceCertSize != ATCATLS_TNGTLS_DEVICE_CERT_SIZE) {
+
+    /* Read device cert signed by the signer above */
+    status = tng_atcacert_read_device_cert(certBuffer, &deviceCertSize,\
+     	                                   &certBuffer[deviceCertSize]);
+    if (ATCA_SUCCESS != status) {
         #ifdef WOLFSSL_ATECC_DEBUG
-        printf("device cert size != ATCATLS_TNGTLS_DEVICE_CERT_SIZE.(%d)\r\n",
-               deviceCertSize);
+        printf("Error reading device cert(0x%x)\r\n", status);
         #endif
-        return WOLFSSL_FAILURE;
+        XFREE(certBuffer,NULL,DYNAMIC_TYPE_TMP_BUFFER);
+        ret = atmel_ecc_translate_err(ret);
+        return ret;
     }
 
-    ret = wolfSSL_CTX_use_certificate_chain_buffer_format(ctx,
-          (const unsigned char*)certBuffer, ATCATLS_TNGTLS_CERT_BUFF_SIZE,
+    ret = wolfSSL_CTX_use_certificate_chain_buffer_format(ctx,\
```

Comment:
Can you remove the trailing `\` on this. Not required for plain C function calls.